cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillRunnerService.java

```diff
@@ -683,20 +683,27 @@ public InetSocketAddress get() {
           // Try to read the port file on the remote host, for up to 3 seconds.
           RetryStrategy strategy = RetryStrategies.timeLimit(3, TimeUnit.SECONDS,
                                                              RetryStrategies.fixDelay(200, TimeUnit.MILLISECONDS));
-          int port = Retries.callWithRetries(() -> {
+          RuntimeMonitorServerInfo serverInfo = Retries.callWithRetries(() -> {
             String json = session.executeAndWait("cat " + programRunId.getRun() + "/"
                                                    + cConf.get(Constants.RuntimeMonitor.SERVER_INFO_FILE));
             RuntimeMonitorServerInfo info = GSON.fromJson(json, RuntimeMonitorServerInfo.class);
             if (info == null) {
               throw new IOException("Cannot find the port for the runtime monitor server");
             }
-            return info.getPort();
+            // This is for backward compatibility case. It could happen when CDAP is upgraded to monitor a
```

Comment:
Aren't programs (such as pipelines) required to stop across CDAP upgrade?
We do document that as a requirement.

Reply:
If there is a run that stuck due to this bug, it can't be stopped. I am not sure if we want to block an upgrade due to this bug.